.github/workflows/build.yml

```diff
@@ -0,0 +1,58 @@
+name: Build
+on:
+  pull_request: {}
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  test:
+    name: Test (${{ matrix.scenario }})
+    runs-on: ubuntu-latest
+    env:
+      CI: 'true'
+    strategy:
+      fail-fast: false
+      matrix:
+        scenario:
+          - default
+          - default-with-lockfile
+          - default-with-jquery
+          - release
+          - beta
+          - canary
+          - lts-3.16
+          - lts-3.12
+          - lts-3.8
+          - lts-3.4
+          - lts-2.18
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Set up Volta
+        uses: volta-cli/action@v1
+      - name: Install dependencies (yarn)
+        if: matrix.scenario == 'default-with-lockfile'
+        run: yarn install --non-interactive --frozen-lockfile
+      - name: Install dependencies (yarn)
+        if: matrix.scenario != 'default-with-lockfile'
+        run: yarn install --non-interactive --no-lockfile
+      - name: Setup ember-try scenario
+        if: matrix.scenario != 'default' && matrix.scenario != 'default-with-lockfile'
```

Comment:
p1/5

FWIW, I find this style significantly harder to read/understand (vs making a different jobs). I'm curious why you prefer it?

Reply:
Mostly because they seem to share a lot of the steps. GH Actions have very limited abstraction/reuse features, so it kind of sucks either way. I am pretty indifferent, if you prefer, I can change it.